gsoc2020/20200303-health-check-wjayesh.md

```diff
@@ -0,0 +1,254 @@
+# UDP(DNS) Health Check for Kubernetes Deployment
+
+| Status        |  Proposed        |
+:-------------- |:---------------------------------------------------- |
+| **RFC #**     | [#3](https://github.com/coredns/rfc/pull/3) |
+| **Author(s)** | Jayesh Sharma (@[wjayesh](https://github.com/wjayesh))  |
+| **Sponsor**   |    |
+| **Updated**   | 2020-03-03                                           |
+| **Obsoletes** |  |
+
+## Objective
+
+CoreDNS is the cluster DNS server for Kubernetes and is very critical for the overall health of the Kubernetes cluster. It is important to monitor the health of CoreDNS itself and restarting or repairing any CoreDNS pods that are not behaving correctly.  
+ 
+While CoreDNS exposes a health check itself in the form of Kubernetes’ `livenessProbe`: 
```

Comment:
@WJayesh , there is a nice trick I learned once. It is called IRAC (Issue, Rule, Application, Conclusion). It might help here. You, by and large, followed it in the "Objective" section. When describing the issue your started to talk about K8s probe.
I would suggest that the issue here is that there is no mechanism to validate that CoreDNS server provides DNS resolution services for entities outside of a specific cluster.

In essence, you are not solving it for Kubernetes only. You are solving it for CoreDNS in general. I really like the reference about the "control plane" 😄 It reminded me about control plane vs data plane issues in hardware routing platforms.

Perhaps, your objective is to build a tool performing check of "DNS resolution plane" and ensuring it performs the functions directed to it by the "control place".

Now, the conversations about the vilability of the `livenessProbe` could be moved to "Monitvation and Scope" section.